prow/cmd/crier/main.go

```diff
@@ -196,13 +200,10 @@ func main() {
 	signal.Notify(sigTerm, syscall.SIGTERM)
 	signal.Notify(sigTerm, syscall.SIGINT)
 
-	go func() {
-		<-sigTerm
-		logrus.Info("Crier received a ternimation signal and is shutting down...")
-		stopCh <- struct{}{}
-		// waiting for all crier worker to finish
-		// TODO(krzyzacy): ideally we know when all worker finishes, rather than blind sleep 5s here
-		time.Sleep(5 * time.Second)
-		logrus.Info("Crier exited")
-	}()
+	<-sigTerm
+	logrus.Info("Crier received a termination signal and is shutting down...")
+	stopCh <- struct{}{}
```

Comment:
Don't you have to send `len(controllers)` objects to the channel?

Reply:
:+1: good catch